File: Leetcode/53.cpp
class Solution {
public:
    int maxSubArray(vector<int> &nums) {
        int curSum = 0, maxAns = -INT_MAX;
        for (auto i : nums) {
            if (curSum < 0) curSum = i;
            else curSum += i;
            maxAns = max(maxAns, curSum);
        }
        return maxAns;
    }
};

File: Leetcode/53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ans, currSum = -10000000000000000000, 0
        for i in nums:
            if currSum < 0:
                currSum = i
            else:
                currSum += i
            ans = max(ans, currSum)

        return ans
